ocaml/xapi/cli_operations.ml

```diff
@@ -2843,15 +2843,34 @@ let vm_migrate printer rpc session_id params =
              let gpu_group = Client.GPU_group.get_by_uuid remote_rpc remote_session gpu_group_uuid in
              vgpu,gpu_group) (read_map_params "vgpu" params) in
 
-         let default_sr =
-           try let pools = Client.Pool.get_all remote_rpc remote_session in
-             printer (Cli_printer.PMsg "Selecting remote pool's default SR for migrating VDIs") ;
-             Some (Client.Pool.get_default_SR remote_rpc remote_session (List.hd pools))
+         let preferred_sr =
+           try
+             let query = Printf.sprintf "(field \"host\"=\"%s\") and (field \"currently_attached\"=\"true\")" (Ref.string_of host) in
+             let host_pbds = Client.PBD.get_all_records_where remote_rpc remote_session query in
+             let srs = List.map (fun (pbd_ref, pbd_rec) -> sr_record remote_rpc remote_session pbd_rec.API.pBD_SR) host_pbds in
+             let (sr, free_space) = List.fold_left (fun (sr, free_space) sr' ->
+               let sr_rec' = sr'.record () in
+               if sr_rec'.API.sR_content_type = "iso" then (sr, free_space)
+               else
+                 let free_space' = Int64.sub sr_rec'.API.sR_physical_size sr_rec'.API.sR_physical_utilisation in
+                 match sr with
```

Comment:
I think this would be a little clearer to read if you would split calculating the free space of an SR and finding the maximum into two separate passes through the srs list (although very slightly less efficient): first you'd `List.map` and calculate the free space like you do here (considering "iso" as `0L`) and return tuples of `(sr_shared, free_space)`.
Then as a second pass you would calculate the maximum with a fold.

Reply:
I agree with @edwintorok. It would also be useful to describe the search policy in a comment, because it is not so obvious what is happening with the `shared` field.

Reply:
I have a slightly different opinion regarding this readability vs efficiency. In this case, the computing of the free_space is actually just one line of code. So pulling out of such logic out from the current fold_left with another fold_left will only slightly enhance the readability of the code in the current loop, but added complexity in the overall code. 

For the concern of the readability, I added some comments in the code, that first introduce the overall principle of the algorithm to determine the SR, and then explained the ISO, shared and available space consideration in details. Please check to see if you are fine with it in the later commit.

Reply:
Actually, the trick of considering the ISO as having zero free-space is not good enough: imagine that the ISO SR is the only shared SR reachable from the destination, then the policy will always select it because shared SRs are preferred. So we must filter out ISOs early. I think Yarsin's structure is fine now.

Reply:
The comment looks good, thanks.